src/library/scala/math/Equiv.scala

```diff
@@ -58,4 +71,438 @@ object Equiv extends LowPriorityEquiv {
     ((x, y) => implicitly[Equiv[S]].equiv(f(x), f(y)))
 
   @inline def apply[T: Equiv]: Equiv[T] = implicitly[Equiv[T]]
+
+  implicit def equivFromPartialOrdering[T](implicit ev: PartialOrdering[T]): Equiv[T] = ev
```

Comment:
This feels a bit contradictory to me... `Ordering[A] <: PartialOrdering[A] <: Equiv[A]`. These implicit conversions already mean that `Equiv` and `PartialOrdering` need to have knowledge about their subtypes. Why not just go down the rabbit hole while we're already stuck halfway, and add all implicit instances of `Equiv` *and its subtypes* to the `Equiv` companion object? 
Also, can someone quickly remind me why `Equiv` needs its own instances for all the primitives and tuples etc?

Reply:
For primitives, it technically doesn't - it can summon those from `Ordering` (that answers the first part of your question I think, kinda). For parameterised types, we can't rely on summoning ones from `Ordering`, because they rely on the existence of implicit *`Ordering`s* for each of the types, when only implicit `Equiv`s should be needed.

We could theoretically not have the instances for primitives, but they are by far *not* the maintenance burden, and it seems a bit odd to me to have methods for `Ordering`, tuples, `Seq`s, etc. but not the primitives. Also, in order to not have confusing deprecation messages, we still need `Equiv` instances for `Float` and `Double`.

Would it be better if we moved the implicit conversions to the subtypes? I think that would still put them on the search path. (i.e. `Ordering` contains `Ordering[A] => PartialOrdering[A]`, and `PartialOrdering` contains `PartialOrdering[A] => Equiv[A]`)